src/main/java/org/jenkinsci/plugins/ewm/strategies/AbstractDiskSpeedStrategy.java

```diff
@@ -0,0 +1,70 @@
+package org.jenkinsci.plugins.ewm.strategies;
+
+import hudson.AbortException;
+import org.jenkinsci.plugins.ewm.DiskAllocationStrategy;
+import org.jenkinsci.plugins.ewm.DiskInfoProvider;
+import org.jenkinsci.plugins.ewm.definitions.Disk;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Base class that selects the {@link Disk} with the highest speed provided by the
+ * {@link #getDiskSpeed(DiskInfoProvider)} method
+ *
+ * @author Alexandru Somai
+ */
+public abstract class AbstractDiskSpeedStrategy extends DiskAllocationStrategy {
+
+    @Nonnull
+    @Override
+    public Disk allocateDisk(@Nonnull List<Disk> disks) throws IOException {
+        long estimatedWorkspaceSize = getEstimatedWorkspaceSize();
+
+        Iterator<Disk> iterator = disks.iterator();
+        Disk candidate = null;
+
+        // try to find at least one Disk candidate that has usable space >= estimated workspace size
+        while (iterator.hasNext()) {
```

Comment:
The code could be refactored a bit. E.g. to a for loop with one check after the run